src/bitboard/mod.rs

```diff
@@ -0,0 +1,1149 @@
+	/// Returns `false` otherwise.
+	pub fn is_full(&self) -> bool {
+		self & &BitBoard::ENDLINE_DELIMITER_MASK == BitBoard::ENDLINE_DELIMITER_MASK
+	}
+
+	/// Returns `true` if **no** bits are set to 1 in the bitboard.
+	/// Returns `false` otherwise.
+	pub fn is_empty(&self) -> bool {
+		for x in &self.b {
+			if *x != 0 {
+				return false;
+			}
+		}
+
+		true
+	}
+
+	/// Returns `true` if **at least** one bit is set to 1 in the bitboard.
+	/// Returns `false` otherwise.
+	pub fn is_any(&self) -> bool {
+		!self.is_empty()
+	}
+
+	pub fn contains_five_aligned(&self) -> bool {
+		for direction in DirectionIterator::new() {
+			let mut tmp = *self;
+			let mut i = 1;
+			while tmp.is_any() {
+				tmp &= tmp << direction;
+				if i >= 5 {
+					return true;
+				}
+				i += 1;
+			}
+		}
+
+		false
+	}
+	// #endregion Test methods
+
+	// ---------------------------------
+	// Computation's method for BitBoard
+	// ---------------------------------
+	pub fn compute_to_isize<F: Fn(&Self) -> isize>(&self, f: F) -> isize {
+		f(self)
+	}
+
+	// ----------
+	// Count bits
+	// ----------
+	pub fn count_ones(&self) -> u16 {
+		self.b
+		    .iter()
+		    .fold(0, |acc: u16, &x| acc + x.count_ones() as u16)
+	}
+
+	// ----------
+	// Bit setter
+	// ----------
+	pub fn set(&mut self, bit_index: isize, bit_value: bool) {
+		let max_index = (BITS_IN_U128 * self.b.len()) as isize;
+		let min_index = -max_index;
+		if bit_index >= max_index || bit_index < min_index {
+			return;
+		}
+		let requested_bit = match bit_index.is_negative() {
+			true => BitBoard::from_array([0, 0, 1]) << (bit_index.abs() - 1) as usize,
+			false => BitBoard::from_array([1 << 127, 0, 0]) >> bit_index as usize,
+		};
+		let is_bit_set = (*self & requested_bit).is_any();
+		let mut new_self = self;
+		if (is_bit_set && !bit_value) || (!is_bit_set && bit_value) {
+			new_self ^= &requested_bit;
+		}
+	}
+
+	// ----------
+	// Bit getter
+	// ----------
+	/// Returns the index of every bit set in the BitBoard in a Vec<u16>.
+	pub fn get_bit_indexes(&self) -> Vec<u16> {
+		let mut result = vec![];
+		let mut index = 0u16;
+		let mut bitboard = *self;
+
+		while bitboard.is_any() {
+			if bitboard.b[0] & Self::U128_FIRST_BIT_SET != 0 {
+				result.push(index);
+			}
+			bitboard = bitboard << 1;
+			index += 1;
+		}
+
+		result
+	}
+
+	// ---------------------------------------
+	// Implementation of bitshift for BitBoard
+	// ---------------------------------------
+	// #region Bitshift implem
+	#[inline]
+	fn shift_left(&self, by: usize) -> Self {
+		let bits = self.b;
+		let max_index = bits.len() - 1;
+		let mut new_bits: [u128; 3] = [0, 0, 0];
+
+		if by >= BITS_IN_U128 * (max_index + 1) {
+			return Self::default();
+		}
+
+		let inner_lshift = by % BITS_IN_U128;
+		let inner_rshift = BITS_IN_U128 - inner_lshift;
+		let value_off = by / BITS_IN_U128;
+		for (dest_i, src_i) in (0..=(max_index - value_off))
+			.rev()
+			.zip((0..=max_index).rev())
+		{
```

Comment:
This feels weird, I would have kept the whole `for` statement on one single line.